src/core/jpx.js

```diff
@@ -24,20 +24,101 @@ class JpxError extends BaseException {
 }
 
 class JpxImage {
-  static #module = null;
-
-  static decode(data, decoderOptions) {
-    decoderOptions ||= {};
-    this.#module ||= OpenJPEG({ warn });
-    const imageData = this.#module.decode(data, decoderOptions);
-    if (typeof imageData === "string") {
-      throw new JpxError(imageData);
+  static #buffer = null;
+
+  static #handler = null;
+
+  static #instantiationFailed = false;
+
+  static #modulePromise = null;
+
+  static #openjpegWasmUrl = "";
+
+  static setOptions({ handler, openjpegWasmUrl }) {
+    if (!this.#buffer) {
+      this.#openjpegWasmUrl = openjpegWasmUrl;
+      if (!openjpegWasmUrl) {
+        this.#handler = handler;
+      }
+    }
+  }
+
+  static async #instantiateWasm(imports, successCallback) {
+    try {
+      if (!this.#buffer) {
+        const wasmFile = "openjpeg.wasm";
+        if (this.#openjpegWasmUrl) {
+          const response = await fetch(`${this.#openjpegWasmUrl}/${wasmFile}`);
```

Comment:
This looks wrong, since these kinds of URLs all should include a trailing slash.
```suggestion
          const response = await fetch(`${this.#openjpegWasmUrl}${wasmFile}`);
```